src/ashell/jerry-code.c

```diff
@@ -32,69 +32,23 @@
 
 static jerry_value_t parsed_code = 0;
 
-typedef struct requires_list {
-    char *file_name;
-    bool reqs_checked;
-    struct requires_list *next;
-    struct requires_list *prev;
-} requires_list_t;
-
-static bool list_contains(requires_list_t **list, const char *file_name)
+static void javascript_print_error(jerry_value_t error_value)
 {
-    requires_list_t *cur = *list;
-    while (cur) {
-        if (strequal(file_name, cur->file_name)) {
-            return true;
-        }
-        cur = cur->next;
-    }
-    return false;
-}
+    if (!jerry_value_has_error_flag(error_value))
+        return;
 
-static void free_list(requires_list_t **list)
-{
-    requires_list_t *cur = *list;
-    while (cur) {
-        zjs_free(cur->file_name);
-        *list = (*list)->next;
-        zjs_free(cur);
-        cur = *list;
-    }
-}
+    jerry_value_clear_error_flag(&error_value);
+    ZVAL err_str_val = jerry_value_to_string(error_value);
 
-static requires_list_t *next_req_to_scan(requires_list_t **list)
-{
-    requires_list_t *cur = *list;
-    while (cur) {
-        if (!cur->reqs_checked) {
-            return cur;
-        } else {
-            cur = cur->next;
-        }
+    jerry_size_t size = 0;
+    char *msg = zjs_alloc_from_jstring(err_str_val, &size);
+    const char *err_str = msg;
+    if (!msg) {
+        err_str = "[Error message too long]";
```

Comment:
This code is duplicate of zjs_util.c's `zjs_print_error_message`; however the string above has [] instead of () so that will be an extra string stored, better to use the same thing for just one copy.

But also we ought to make this common code somewhere and reuse it instead of duplicating.